topeft/modules/datacard_tools.py

```diff
@@ -847,22 +847,43 @@ def analyze(self,km_dist,ch,selected_wcs, crop_negative_bins):
                         "rate": -1
                     }
                     # There should be only 1 sparse axis at this point, the systematics axis
+                    check_zero_arr0 = False
+                    check_zero_arr1 = False
                     for sp_key,arr in v.items():
+                        syst = sp_key[0]
                         if crop_negative_bins:
                             negative_bin_mask = np.where( arr[0] < 0) # see where bins are negative
                             arr[0][negative_bin_mask] = np.zeros_like( arr[0][negative_bin_mask] )  # set those to zero
                             if arr[1] is not None:
                                 arr[1][negative_bin_mask] = np.zeros_like( arr[1][negative_bin_mask] )  # if there's a sumw2 defined, that one's set to zero as well. Otherwise we will get 0 +/- something, which is compatible with negative
 
-                        syst = sp_key[0]
+                        if len(arr[0][negative_bin_mask]) and syst =="nominal":  # check systematics error for fake factors
+                            if sum(arr[0]) == 0:
+                                check_zero_arr0 = True
+                            if sum(arr[1]) == 0:
+                                check_zero_arr1 = True
+                        if "Up" in syst:
```

Comment:
Thanks @ywan2 for making this change! You could probably remove
`if "Down" in syst`
and change this to
`if "Up" in syst or "Down" in syst`
or even
`if syst != "nominal"`